test/oas2json.test.js

```diff
@@ -32,100 +32,169 @@ tap.test('oas2json', async t => {
-              items: {
-                $ref: 'Tag.json'
+            title: 'Pet',
+            $id: 'Pet.json'
+          },
+          'Pet.json schema is created correctly'
+        )
+        const customerSchema = fs.readJsonSync(
+          resolveFromPackageRoot(schemasDir, 'Customer.json')
+        )
+        t.same(
+          customerSchema,
+          {
+            type: 'object',
+            properties: {
+              id: {
+                type: 'integer',
+                format: 'int64',
+                minimum: -9223372036854776000,
+                maximum: 9223372036854776000
+              },
+              username: {
+                type: 'string'
+              },
+              address: {
+                type: 'array',
+                items: {
+                  $ref: 'Address.json'
+                }
               }
             },
-            status: {
-              type: 'string',
-              description: 'pet status in the store',
-              enum: ['available', 'pending', 'sold']
-            }
+            title: 'Customer',
+            $id: 'Customer.json'
           },
-          title: 'Pet',
-          $id: 'Pet.json'
-        },
-        'Pet.json schema is created correctly'
+          'Customer.json schema is created correctly'
+        )
+      }
+    )
+
+    t.test('should convert components.schemas automatically', async t => {
+      runCommand(inputPath, outputPath, 'components.requestBodies')
+
+      const generatedDirs = fs.readdirSync(outputPath)
+      t.match(
+        generatedDirs,
+        ['components.requestBodies', 'components.schemas'],
+        'generates the expected directories'
       )
-      const customerSchema = fs.readJsonSync(
-        resolveFromPackageRoot(outputPath, 'Customer.json')
+
+      const petReqBody = fs.readJSONSync(
+        resolveFromPackageRoot(outputPath, 'components.requestBodies/Pet.json')
       )
       t.same(
-        customerSchema,
+        petReqBody,
         {
-          type: 'object',
-          properties: {
-            id: {
-              type: 'integer',
-              format: 'int64',
-              minimum: -9223372036854776000,
-              maximum: 9223372036854776000
-            },
-            username: {
-              type: 'string'
+          description: 'Pet object that needs to be added to the store',
+          content: {
+            'application/json': {
+              schema: {
+                $ref: 'Pet.json'
+              }
             },
-            address: {
-              type: 'array',
-              items: {
-                $ref: 'Address.json'
+            'application/xml': {
+              schema: {
+                $ref: 'Pet.json'
               }
             }
           },
-          title: 'Customer',
-          $id: 'Customer.json'
+          title: 'Pet',
+          $id: 'Pet.json'
         },
-        'Customer.json schema is created correctly'
+        'generates the expected JSON'
```

Comment:
(Related to the comment above about `fromSchema`)

Most of OpenApi def translates 1o1 to a JSON schema definitions since the 2 formats are almost identical except for a few minor things. One of the is the OpenAPI `nullable` prop which needs conversion.

```
    Answer:
      type: string
      nullable: true
      enum:
        - yes
        - no
```

I'd suggest to review the examples to use OpenApi definitions which undergo some transformation in the JSON schema conversion, so that we can make sure that we are actually convertig.